File: programmers/[PCCP 기출문제] 1번 동영상 재생기.py
def toStrUnder10(n):
    if n < 10:
        n = '0' + str(n)
    else:
        n = str(n)
    return n


def solution(video_len, pos, op_start, op_end, commands):
    # 다 초 단위로 변환
    arr = video_len, pos, op_start, op_end
    arrL = []
    arrI = []
    for a in arr:
        arrL.append(list(map(int, a.split(':'))))
    for a in arrL:
        arrI.append(a[0] * 60 + a[1])

    video_lenI = arrI[0]
    posI = arrI[1]
    op_startI = arrI[2]
    op_endI = arrI[3]

    # 오프닝 건너뛰기 자동 실행
    if op_startI <= posI <= op_endI:
        posI = op_endI

    # commands -> 하나씩 하는 게 아니라 다 합해서 하면 됨 -> xx 0초, 마지막초 때문에 안됨. 하나씩 해야함
    for c in commands:
        if c == 'prev':
            if posI < 10:
                posI = 0
            else:
                posI -= 10
        elif c == 'next':
            if posI > video_lenI - 10:
                posI = video_lenI
            else:
                posI += 10
        # 오프닝 건너뛰기 자동 실행
        if op_startI <= posI <= op_endI:
            posI = op_endI

    minute = toStrUnder10(posI // 60)
    second = toStrUnder10(posI % 60)

    answer = minute + ':' + second
    return answer

File: programmers/[PCCP 기출문제] 1번 동영상 재생기.js
const paserTimeToNum = (arr) => {
    return Number(arr.split(':')[0]) * 60 + Number(arr.split(':')[1])
}

function solution(video_len, pos, op_start, op_end, commands) {
    let video_t = paserTimeToNum(video_len)
    let pos_t = paserTimeToNum(pos)
    let op_start_t = paserTimeToNum(op_start)
    let op_end_t = paserTimeToNum(op_end)
    
    for (const command of commands) {
        if ((pos_t >= op_start_t) && (pos_t <= op_end_t)) {
            pos_t = op_end_t
        }
        if (command === 'prev') {
            pos_t = pos_t - 10 < 0 ? 0 : pos_t - 10
        } else {
            pos_t = pos_t + 10 > video_t ? video_t : pos_t + 10
        }
    }
    if ((pos_t >= op_start_t) && (pos_t <= op_end_t)) {
        pos_t = op_end_t
    }
    
    const s = pos_t % 60
    const m = (pos_t - s) / 60
    
    let mm = m < 10 ? '0' + m.toString() : m.toString()
    let ss = s < 10 ? '0' + s.toString() : s.toString()
    return mm + ':' + ss;
}